include/zephyr/net/wifi.h

```diff
@@ -86,6 +88,10 @@ enum wifi_security_type {
 	WIFI_SECURITY_TYPE_FT_EAP,
 	/** FT-EAP-SHA384 security */
 	WIFI_SECURITY_TYPE_FT_EAP_SHA384,
+	/** WPA/WPA2/WPA3 PSK security, SAE with hunting-and-pecking loop. */
+	WIFI_SECURITY_TYPE_WPA_AUTO_PERSONAL_HNP,
```

Comment:
Can you please update commit log as to why this was necessary? Is there a specific reason or issue this solves?

Reply:
For mix mode connection, need set pwe type for SAE to compatible with ext-AP, will connect failed if not compatible.

Reply:
Updated commit info.

Reply:
but in AUTO mode we do enable both HnP and H2E, so, that should work with any AP config, no?

Reply:
Yes, but AUTO mode mapping to mix mode with sae pwe type 0 if the impl without this change.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hostap/src/supp_api.c#L2330 it's 2.

Reply:
yes, sorry my mistake, what I want to express is the original impl just cover only one sae pwe type.

Reply:
`2` mean both HnP and H2E. Can you please share the problem that this PR is solving?

Reply:
Yes, I got your point, I tried to get history of the issue related to this PR, but only get info that if not configure pwe as 2 for mix mode, connect failed with AP include 0x32 IE that specify bss require H2E.

1. Without this change, I test mix mode with AP include 0x32 IE, connect success.
2. And mix mode should compatible all sae pwe type, so the current impl(mix mode set pwe as 2) is suitable.

I think should remove the change about mix mode of this PR. What do you think about it? @krish2718 
Avoid PR merged, maochen help to revert the approve.

Reply:
Yes, IMHO dropping https://github.com/zephyrproject-rtos/zephyr/pull/82956/commits/3d57b012c2c1f2ad31612cfea4fb7a84977b4ac0 should be fine, remaingin two commits are for SAP.

Reply:
Updated as comment.